File: src/main/java/io/sch/historyscan/infrastructure/logging/AppLogger.java
package io.sch.historyscan.infrastructure.logging;

import io.sch.historyscan.domain.logging.spi.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class AppLogger implements Logger {
    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(AppLogger.class);

    @Override
    public void info(String message) {
        logger.info(message);
    }

    @Override
    public void error(Exception ex) {
        logger.error(ex.getMessage(), ex);
    }

    @Override
    public void error(String message) {
        logger.error(message);
    }

    @Override
    public void debug(String message) {
        logger.debug(message);
    }

    @Override
    public void error(String message, Exception exception) {
        logger.error(message, exception);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/logging/AppLogger.kt
package io.sch.historyscan.infrastructure.logging

import io.sch.historyscan.domain.logging.spi.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class AppLogger : Logger {
    override fun info(message: String) {
        logger.info(message)
    }

    override fun error(ex: Exception) {
        logger.error(ex.message, ex)
    }

    override fun error(message: String) {
        logger.error(message)
    }

    override fun debug(message: String) {
        logger.debug(message)
    }

    override fun error(message: String, exception: Exception) {
        logger.error(message, exception)
    }

    companion object {
        private val logger: org.slf4j.Logger = LoggerFactory.getLogger(AppLogger::class.java)
    }
}
